std/meta.d

```diff
@@ -267,12 +267,23 @@ if (!isAggregateType!T || is(Unqual!T == T))
 
 /**
  * Returns the index of the first occurrence of type T in the
- * sequence of zero or more types TList.
+ * sequence of zero or more types Seq.
  * If not found, -1 is returned.
  */
-template staticIndexOf(alias T, TList...)
+template staticIndexOf(alias T, Seq...)
 {
-    enum staticIndexOf = genericIndexOf!(T, TList);
+    static foreach (idx, arg; Seq)
+    {
+        static if (is(typeof(staticIndexOf) == void) && // not yet defined
+                   isSame!(T, arg))
+        {
+            enum staticIndexOf = idx;
+        }
```

Comment:
Simply `break` out of the loop?

```suggestion
    Loop:
    static foreach (idx, arg; Seq)
    {
        static if (isSame!(T, arg))
        {
            enum staticIndexOf = idx;
            break Loop;
        }
```

Reply:
@MoonlightSentinel:
From https://dlang.org/spec/version.html#break-continue:
> Note that static foreach itself cannot be broken nor continued even if it is explicitly labeled

(I haven't actually tried it recently though). Also I didn't write the code, the diff just makes it look like that. I'm just renaming some identifiers ;-)

Reply:
You're right, I misremembered the interaction between `static foreach` and `break`.

Reply:
Im curious how this implementation compares to something like this:

```
template staticIndexOf(alias T, Seq...)
{
    enum staticIndexOf = ()
    {
        foreach(idx, arg; Seq)
        {
            if (isSame!(T, arg))
                return idx;
        }
        
        return -1;
    }();
}
```

Reply:
Interesting solution. I don't know, but for large sequence length I think static foreach is v slow, at least ATM:
http://dpldocs.info/this-week-in-d/Blog.Posted_2020_05_11.html#dmd-and-static-foreach

Reply:
That instantiates `isSame` for every element of `Seq`, which could be expensive if there's a match near the start.